File: Scala3/src/test/scala/lexer/ScannerTest.scala
package lexer

import org.junit.Assert.*
import org.junit.Test

class ScannerTest:
  @Test def run(): Unit =
    val sut = Scanner("123")

    assertFalse(sut.isEot())
    assertEquals(Some('1'), sut.peek())
    assertEquals(Some('1'), sut.consume())

    assertFalse(sut.isEot())
    assertEquals(Some('2'), sut.peek())
    assertEquals(Some('2'), sut.consume())

    assertFalse(sut.isEot())
    assertEquals(Some('3'), sut.peek())
    assertEquals(Some('3'), sut.consume())

    assertTrue(sut.isEot())
    assertEquals(None, sut.peek())
    assertEquals(None, sut.consume())

  @Test def isEot(): Unit =
    val sut = Scanner("")

    assertTrue(sut.isEot())

File: Java/app/src/test/java/com/github/shin1x1/jsonparser/lexer/ScannerTest.java
package com.github.shin1x1.jsonparser.lexer;

import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.*;

class ScannerTest {
    @Test
    void peek() throws IOException {
        var sut = new Scanner("1");
        assertEquals('1', sut.peek().orElseThrow());
    }

    @Test
    void consume() throws IOException {
        var sut = new Scanner("1あ3");
        assertEquals('1', sut.consume().orElseThrow());
        assertEquals('あ', sut.consume().orElseThrow());
        assertEquals('3', sut.consume().orElseThrow());
        assertTrue(sut.consume().isEmpty());
    }

    @Test
    void isEot() throws IOException {
        var sut = new Scanner("1");
        assertFalse(sut.isEot());
        sut.consume();
        assertTrue(sut.isEot());
    }
}